lib/tools/adb-emu-commands.js

```diff
@@ -5,42 +5,55 @@ import path from 'path';
 
 let emuMethods = {};
 
+emuMethods.getEmulators = async function() {
+  if (_.isArray(this.emulators) &&
+      this.emulators.length) {
+    return this.emulators;
+  }
+  this.emulators = await this.getConnectedEmulators();
+  return this.emulators;
+};
+
 emuMethods.isEmulatorConnected = async function (udid) {
-  let emulators = await this.getConnectedEmulators();
+  let emulators = await this.getEmulators();
   if (!emulators.length) {
     return false;
   }
-  for (let emulator of emulators) {
-    if (emulator.udid === udid) {
-      return true;
+  return !!_.find(emulators, (x) => x && x.udid === udid);
+};
+
+emuMethods.checkEmulatorConnected = async function (udid) {
```

Comment:
I'm seeing that we dont set the udid in the `appium-android-driver` call https://github.com/appium/appium-android-driver/blob/e073b913f3deefdb94cb3ea0a4c9e6f9721b0a0b/lib/commands/actions.js#L179-L181 and the devideId is already set after prepareEmulator https://github.com/appium/appium-android-driver/blob/master/lib/android-helpers.js#L119 so I guess we can remove the udid validation. what do you think @jlipps @mykola-mokhnach ?